sapmon/payload/netweaver/rfcclient.py

```
@@ -1279,3 +1406,81 @@ def _decorateLockMetrics(self, records: list) -> None:
+                                     startDateTime: datetime,
+                                     endDateTime: datetime,
+                                     logTag: str):
+        rfcName = '/SAPDS/RFC_READ_TABLE2'
+        self.tracer.info(("[%s] invoking rfc %s for hostname=%s for client %s"),
+                         logTag, 
+                         rfcName, 
+                         self.sapHostName,
+                         self.sapClient)
+
+        try:
+            # passing OPTIONS table as a input parameter to the RFC call which accepts input in the following format
+            # AS4DATE BETWEEN 20211010 AND 20211015
+            options_table = ('AS4DATE BETWEEN %s AND %s'% ("".join(str(startDateTime.date()).split("-")), "".join(str(endDateTime.date()).split("-"))))
+            stms_records = connection.call(rfcName,
+                                                  QUERY_TABLE = 'E070',
+                                                  DELIMITER = ';',
+                                                  OPTIONS = [options_table])
+            return stms_records
+
+        except ABAPRuntimeError as e:
+            self.tracer.error("[%s] Runtime error for rfc %s with hostname: %s (%s). Update the roles in SAP System using role file %s",
+                              logTag, rfcName, self.sapHostName, e, self.rolesFileURL, exc_info=True)
+        except Exception as e:
+            self.tracer.error("[%s] Error occured for rfc %s with hostname: %s (%s)", 
+                              logTag, rfcName, self.sapHostName, e, exc_info=True)
+
+        return None
+
+    """
+    parse results from /SAPDS/RFC_READ_TABLE2 and enrich with additional properties
+    """
+    def _parseChangeAndTransportResult(self, rfcName, result, logTag):
+        if result is None:
+            raise ValueError("%s empty result received for /SAPDS/RFC_READ_TABLE2 RFC from hostname: %s"
+                             % (logTag, self.sapHostName))
```

I thought we weren't doing this anymore with RFC calls?  In other words, I thought we wanted the RFC method itself to throw an exception if the response was invalid for any reason, instead of return None.

And then the _parse method should be able to handle an empty response if it is a legitimately empty response case, as in the call was successful but no data was returned.  If the call was not successful for any reason, then exception should bubble up.  The only edge case is when the RFC call can legitimately throw an exception to signify no data was returned, and that we want to treat this as a valid request.